datafusion/core/src/execution/context/mod.rs

```diff
@@ -476,6 +476,28 @@ impl SessionContext {
         self.execute_logical_plan(plan).await
     }
 
+    /// Creates logical expresssions from SQL query text.
+    ///
+    /// # Example: Parsing SQL queries
+    ///
+    /// ```
+    /// # use std::sync::Arc;
+    /// # use arrow::datatypes::{DataType, Field, Schema};
+    /// # use datafusion::prelude::*;
+    /// # use datafusion_common::DFSchema;
+    /// let sql = "SELECT a FROM t WHERE b = 1";
+    /// // provide type information that `a` is an Int32
+    /// let schema = Schema::new(vec![Field::new("a", DataType::Int32, true)]);
+    /// let df_schema = DFSchema::try_from(schema).unwrap();
+    ///
+    /// let expr = SessionContext::new()
+    ///  .parse_sql(sql, &df_schema)
+    ///  .await?;
+    /// ```
+    pub async fn parse_sql(&self, sql: &str, df_schema: &DFSchema) -> Result<Expr> {
```

Comment:
maybe "parse_sql_expr" would be a better name as it may not be clear statements like `SELECT ...` aren't supported

Reply:
Yeah, initially I thought this was a feature, but they are not the same.

https://github.com/apache/datafusion/blob/5bfc11ba4ac4f11eaf9793c668e4a064fb697e6e/datafusion/core/src/execution/context/mod.rs#L438

Done.